From 77de7ab7080299c2c7512dad02d1dc49dd94df4a Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Tue, 13 Aug 2024 22:42:49 +0530
Subject: [PATCH] meson: Downgrade MSRV to 1.77

We can't use 1.78 or newer due to an MSVC static linking regression:

https://github.com/rust-lang/rust/issues/129020

https://github.com/rust-lang/cc-rs/issues/1167#issuecomment-2281193970
---
 Cargo.toml  | 2 +-
 meson.build | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index ef03c645..0c59ea91 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -12,7 +12,7 @@ repository = "https://gitlab.gnome.org/GNOME/librsvg/"
 categories = ["graphics", "multimedia::images", "rendering::data-formats"]
 keywords = ["svg", "cairo"]
 edition = "2021"
-rust-version = "1.78.0"
+rust-version = "1.77.0"
 
 [profile.overflow]
 inherits = "release"
diff --git a/meson.build b/meson.build
index 685f526d..cdcc929d 100644
--- a/meson.build
+++ b/meson.build
@@ -23,7 +23,7 @@ cc = meson.get_compiler('c')
 #   - the "rust-version" value in Cargo.toml
 #   - the "_manual_setup" section of devel-docs/devel_environment.rst
 #   - the "RUST_MINIMUM" variable in ci/container_builds.yml
-msrv = '1.78.0'
+msrv = '1.77.0'
 cargo = find_program('cargo', version:'>= @0@'.format(msrv))
 cargo_wrapper = find_program('meson/cargo_wrapper.py', native: true)
 cargo_c = find_program('cargo-cbuild', version:'>= 0.9.19')
-- 
2.46.0

